projects/dxilconv/lib/DxbcConverter/DxbcConverterImpl.h

```diff
@@ -407,12 +407,9 @@ class DxbcConverter : public IDxbcConverter {
     };
     vector<pair<unsigned, BasicBlock*> > SwitchCases;  // Switch
 
-    Scope() : Kind(Kind::Function), pPreScopeBB(nullptr), pPostScopeBB(nullptr), NameIndex(0), 
-              pThenBB(nullptr), pElseBB(nullptr), pCond(nullptr),
-              pLoopBB(nullptr), ContinueIndex(0), LoopBreakIndex(0),
-              pDefaultBB(nullptr), pSelector(nullptr), CaseGroupIndex(0), SwitchBreakIndex(0),
-              LabelIdx(0), CallIdx(0), ReturnTokenOffset(0), ReturnIndex(0), bEntryFunc(false),
-              pHullLoopBB(nullptr), HullLoopBreakIndex(0), pInductionVar(nullptr), HullLoopTripCount(0) {}
+    Scope() : Kind(Kind::Function), pPreScopeBB(nullptr), pPostScopeBB(nullptr), NameIndex(0) {
+      memset(reinterpret_cast<char*>(&pThenBB), '\0', reinterpret_cast<char*>(&SwitchCases) - reinterpret_cast<char*>(&pThenBB));
```

Comment:
This seems pretty fragile, but I guess the other options (while keeping this completely NFC, anyway) aren't really better

Reply:
Yea... The thing that makes this really gross to me is that since the union and contained structs are anonymous you can't just take the sizeof... instead you have to do member offset shenanigans.

I could not think of a better way to accomplish this without just rewriting the code.